compliance/src/main/java/bio/terra/pearl/compliance/SyncVantaUsers.java

```diff
@@ -0,0 +1,296 @@
+import io.cloudevents.CloudEvent;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.Banner;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.WebApplicationType;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.util.retry.Retry;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Run this as a main() or as a cloud function to set
+ * various vanta integrations to use only the team members
+ * needed for our audit.  When running locally, set the
+ * GOOGLE_CLOUD_PROJECT and VANTA_CONFIG_SECRET environment vars.
+ */
+@Slf4j
+@SpringBootApplication
+public class SyncVantaUsers implements CommandLineRunner, CloudEventsFunction {
+
+    @Autowired
+    private SecretManagerTemplate secretManagerTemplate;
+
+    private UserSyncConfig userSyncConfig;
+
+    @Value("#{environment.VANTA_CONFIG_SECRET}")
+    private String vantaConfigSecret;
+
+    private String vantaBasePath;
+
+    private String vantaClientId;
+
+    private String vantaSecret;
+
+    private final Collection<PersonInScope> peopleInScope = new ArrayList<>();
+
+    private final SpringApplication app = new SpringApplicationBuilder(SyncVantaUsers.class).web(WebApplicationType.NONE).bannerMode(Banner.Mode.OFF).build();
+
+    @Override
+    public void accept(CloudEvent event) throws Exception {
+        try {
+            app.run("");
+        } catch (Exception e) {
+            log.error("Vanta sync failed", e);
+        }
+    }
+
+    public static void main(String[] args) {
+        try {
+            new SyncVantaUsers().accept(null);
+        } catch (Exception e) {
+            log.error("Could not run sync", e);
+        }
+    }
+
+    @Override
+    public void run(String... args) throws Exception {
+        userSyncConfig = new Gson().fromJson(secretManagerTemplate.getSecretString(vantaConfigSecret), UserSyncConfig.class);
+        vantaBasePath = userSyncConfig.getVantaBaseUrl();
+        vantaClientId = userSyncConfig.getVantaClientId();
+        vantaSecret = userSyncConfig.getVantaClientSecret();
+        peopleInScope.addAll(userSyncConfig.getPeopleInScope());
+        Instant start = Instant.now();
```

Comment:
```suggestion
       
       // now that we've loaded the config, perform the sync 
        Instant start = Instant.now();
```

Reply:
also, FYI, Spring has a StopWatch class that's intended for measuring execution time like this.